src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -675,26 +564,43 @@ def sync_lock(self) -> Lock:
 
     def _initialize(self) -> None:
         """Initialize the SQL store."""
-        from zenml.zen_stores.metadata_store import MetadataStore
-
         logger.debug("Initializing SqlZenStore at %s", self.config.url)
 
-        metadata_config = self.config.get_metadata_config()
-        self._metadata_store = MetadataStore(config=metadata_config)
-
         url, connect_args, engine_args = self.config.get_sqlmodel_config()
         self._engine = create_engine(
             url=url, connect_args=connect_args, **engine_args
         )
+
+        # SQLite: As long as the parent directory exists, SQLAlchemy will
+        # automatically create the database.
+        local_path = self.config.get_local_path(url)
+        if local_path and not fileio.exists(local_path):
+            fileio.makedirs(os.path.dirname(local_path))
+
+        # MySQL: We might need to create the database manually.
+        # To do so, we create a new engine that connects to the `mysql` database
+        # and then create the desired database.
+        # See https://stackoverflow.com/a/8977109
+        if (
+            self.config.driver == SQLDatabaseDriver.MYSQL
+            and self.config.database
+        ):
+            master_url = self._engine.url._replace(database="mysql")
+            master_engine = create_engine(
```

Comment:
A good ops practice is to manually create the database and then restrict the DB user permissions to only have access to that database. In that case, I imagine that trying to access the mysql database will fail.
You should first try to find out if the database exists in the context of the current user.